File: Interview/509/fib.py
class Solution:
    def fib(self, n: int) -> int:
        if n==0:
            return 0
        elif n==1:
            return 1
        
        second_last=0
        last=1
        cur=1
        
        for i in range(2,n+1):
            cur=second_last+last
            second_last=last
            last=cur
        
        return cur

File: Coding/509/fib.cpp
class Solution {
public:
    int fib(int n) {
        int last = 1;
        int second_last = 0;
        int cur = 0;
        
        // base case
        if (n==0) {
            return 0;
        } else if (n==1) {
            return 1;
        }
        
        // iteration
        for(int i = 2;i<=n;i++){
            cur = last + second_last;
            second_last = last;
            last = cur;
        }
        
        return cur;
    }
};
